tools/palletjack2unbound

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env ruby
+
+# Write DNS resolver stub declarations from a Palletjack warehouse
+
+require 'palletjack'
+require 'optparse'
+require 'dns/zone'
+require 'ip'
+
+$options = {}
+
+def exists_ok(&code)
+  code.call()
+rescue Errno::EEXIST
+  nil
+end
+
+def stub_zone(zone, *stub_addrs)
+  head = "################################################################
+# Automatically generated by palletjack2unbound from 
+# #{$options[:warehouse]}
+################################################################
+
+stub-zone:
+  name: #{zone}\n"
+
+  stub_addrs.inject(head) do |result, addr|
+    result << "  stub-addr: #{addr}\n"
+  end
+end
+
+opts = OptionParser.new
+opts.banner = "Usage: #{$PROGRAM_NAME} -w <warehouse> -d <output directory>
+
+Write DNS resolver stub declarations from a Palletjack warehouse
+into Salt state configuration for unbound.
+
+E.g.
+  palletjack2unbound -w /etc/salt/respository/warehouse \\
+    -o /etc/salt/repository/state/unbound/files
+"
+opts.on("-w DIR", "--warehouse DIR", "warehouse directory", String) {|dir|
+  $options[:warehouse] = dir }
+opts.on("-o DIR", "--output DIR", "output directory", String) {|dir|
+  $options[:output] = dir
+  $options[:conf_dir] = "#{dir}/conf.d"
+}
+opts.parse!
+
+if not $options[:warehouse] or
+    not $options[:output] or
+    not File.directory?($options[:output])
+  puts opts.to_s
+  exit 1
+end
+
+exists_ok { Dir.mkdir($options[:conf_dir]) }
+
+jack = PalletJack.new($options[:warehouse])
+
+jack["domain"].each do |domain|
+  zone = "#{domain['net.dns.domain']}"
+
+  stub_addrs = []
+  domain['net.dns.ns'].each do |ns|
+    jack["ipv4_interface", with_all:{"net.dns.fqdn" => ns}].each do |ipv4|
```

Comment:
I would rather have an `ipv4_address` abstraction than `ipv4_interface`, but that re-factoring work is out of scope for this. If the warehouse does not contain the IPv4 address for the NS, we can't produce `stub-zone` glue, and the resolver must rely on normal recursion from root hints.